src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/services/CheckpointService.scala

```diff
@@ -104,23 +109,29 @@ class CheckpointServiceImpl @Inject() (userGroupInformationWrapper: UserGroupInf
    *  @tparam R type of the parsed value
    *  @return None if the file doesn't exist, Some with the parsed content
    */
-  private def parseFileAndClose[R](pathStr: String, parseFn: Iterator[String] => R): Option[R] = {
+  private def parseFileAndClose[R](pathStr: String, parseFn: Iterator[String] => R)(implicit
+    ugi: UserGroupInformation
+  ): Try[Option[R]] = {
     val path = new Path(pathStr)
-    if (fs.exists(path)) {
-      val input = fs.open(path)
-      try {
-        val lines = Source.fromInputStream(input, UTF_8.name()).getLines()
-        Some(parseFn(lines))
-      } catch {
-        case e: Exception =>
-          // re-throw the exception with the log file path added
-          throw new Exception(s"Failed to parse file $path", e)
-      } finally {
-        IOUtils.closeQuietly(input)
+    hdfsService.exists(path).flatMap { exists =>
+      if (exists) {
+        hdfsService.open(path).map { input =>
+          try {
+            val lines = Source.fromInputStream(input, UTF_8.name()).getLines()
```

Comment:
I would leave open and read on hdfs service -> hdfsService.parseFileAndClose

Reply:
Ok, yes makes sense